torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -139,6 +141,14 @@ std::vector<at::Device> getDeviceList(const std::vector<at::Tensor>& tensors) {
   return res;
 }
 
+// Return CUDA device with ordinal given by input rank.
+at::Device getDeviceForRank(int rank) {
+  TORCH_CHECK(rank >= 0, "Invalid rank ", rank);
```

Comment:
This is mostly here for the test - to simulate the case where an exception (rather than timeout) happens in health check.